Data Engineer Assessment 

INCUBYTES

Tisha Goyal

In [1]:
import pandas as pd

In [2]:
z=[]
with open('Document.txt','r') as file:
    col= file.readline().strip().lstrip("|")
    for i in file:
        z.append(i.strip().lstrip("|"))
        
#print(col)    
#print(z)


In [3]:
x= col.split("|")
x  
df= pd.DataFrame(columns= [x])
df

Empty DataFrame
Columns: [(H,), (Customer_Name,), (Customer_Id,), (Open_Date,), (Last_Consulted_Date,), (Vaccination_Id,), (Dr_Name,), (State,), (Country,), (DOB,), (Is_Active,)]
Index: []

In [4]:
temp=[]

for i in range(len(z)):
    temp=(z[i].split("|"))
    df.loc[len(df)] = temp
    print(temp)

['D', 'Alex', '123457', '20101012', '20121013', 'MVD', 'Paul', 'SA', 'USA', '06031987', 'A']
['D', 'John', '123458', '20101012', '20121013', 'MVD', 'Paul', 'TN', 'IND', '06031987', 'A']
['D', 'Mathew', '123459', '20101012', '20121013', 'MVD', 'Paul', 'WAS', 'PHIL', '06031987', 'A']
['D', 'Matt', '12345', '20101012', '20121013', 'MVD', 'Paul', 'BOS', 'NYC', '06031987', 'A']
['D', 'Jacob', '1256', '20101012', '20121013', 'MVD', 'Paul', 'VIC', 'AU', '06031987', 'A']
['D', 'Alex', '123457', '20101012', '20121013', 'MVD', 'Paul', 'SA', 'IND', '06031987', 'A']
['D', 'John', '123458', '20101012', '20121013', 'MVD', 'Paul', 'TN', 'USA', '06031987', 'A']


In [9]:

column_names= ["H", "Customer_Id", "Customer_Name", "Open_Date", "Last_Consulted_Date", "Vaccination_Id",  "Dr_Name", "State", "Country", "DOB", "Is_Active"]
df = df.reindex(columns= [column_names])
df= df.drop(columns=["H"], level = 0)
df


Customer_Id Customer_Name Open_Date Last_Consulted_Date Vaccination_Id  \
0      123457          Alex  20101012            20121013            MVD   
1      123458          John  20101012            20121013            MVD   
2      123459        Mathew  20101012            20121013            MVD   
3       12345          Matt  20101012            20121013            MVD   
4        1256         Jacob  20101012            20121013            MVD   
5      123457          Alex  20101012            20121013            MVD   
6      123458          John  20101012            20121013            MVD   

  Dr_Name State Country       DOB Is_Active  
0    Paul    SA     USA  06031987         A  
1    Paul    TN     IND  06031987         A  
2    Paul   WAS    PHIL  06031987         A  
3    Paul   BOS     NYC  06031987         A  
4    Paul   VIC      AU  06031987         A  
5    Paul    SA     IND  06031987         A  
6    Paul    TN     USA  06031987         A

This data can be transfered to the database now

To group data according Country we can use the following query in SQL

>> SELECT *  FROM table_name GROUP BY Country

Future to retirve data from the tables using Customer_id
we can follow the following queries 

>>SELECT * FROM table_name WHERE Customer_Id = "123457"